### PR TITLE
Dashboard: Move story duplication to the server

### LINKED ISSUE #8664: Dashboard: Move story duplication to the server
## Feature Description

Currently in the dashboard we have to pass both `story_data` and `content` for duplicating stories, which means we need to retrieve these fields all the time:

https://github.com/google/web-stories-wp/blob/c6432948c1a91f42ca19b1069255281fff43dd0a/packages/dashboard/src/app/api/useStoryApi.js#L312-L369

Moving this to the server would reduce response sizes and thus performance on the dashboard.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

* No more `story_data` and `content` fields being sent & preloaded
* Duplication still works

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Copy stories at server level and remove need to need story_data and content. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Login as admin
2. Create a story with some pages. 
3. Go back to dashboard.
4. Duplicate story. 
5. Validate all data is copied over.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.  

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8664

### CODE REVIEW

File: includes/Admin/Dashboard.php [hunk 1]
```diff
@@ -265,10 +265,6 @@ public function load_stories_dashboard() {
 								'featured_media_url',
 								'preview_link',
 								'edit_link',
-								// TODO: Remove need for this as it's a lot of data sent over the wire.
-								// It's only needed for duplicating stories.
-								'content',
-								'story_data',
 								// _web_stories_envelope will add these fields, we need them too.
```
Comment (BrittanyIRL) [inline includes/Admin/Dashboard.php:268]: 🎉

File: packages/dashboard/src/app/api/useStoryApi.js [hunk 2]
```diff
@@ -127,33 +121,17 @@ const useStoryApi = (dataAdapter, { storyApi, encodeMarkup }) => {
         }
         isInitialFetch.current = false;
 
-        // Hardening in case data returned by the server is malformed.
-        // For example, story_data could be missing/empty.
-        const cleanStories = response.body.filter((story) =>
-          Array.isArray(story?.story_data?.pages)
-        );
+        const stories = response.body;
```
Comment (BrittanyIRL) [inline packages/dashboard/src/app/api/useStoryApi.js:129]: This is no longer an edge case?  https://github.com/google/web-stories-wp/pull/6095

Comment (spacedmonkey) [reply packages/dashboard/src/app/api/useStoryApi.js:129]: The field is not longer in the api to make responses faster. We also do not use that story data for anything anymore. So safe to remove.

**Review by BrittanyIRL — APPROVED** [review]
Tested locally works great, LGTM.

File: includes/REST_API/Stories_Base_Controller.php [hunk 3]
```diff
@@ -147,6 +147,58 @@ public function prepare_item_for_response( $post, $request ) {
...
+		}
+
+		$original_post = $this->get_post( (int) $original_id );
+		if ( is_wp_error( $original_post ) ) {
+			return $original_post;
+		}
+
+		if ( ! $this->check_read_permission( $original_post ) ) {
+			return new WP_Error(
+				'rest_cannot_view',
```
Comment (swissspidy) [inline includes/REST_API/Stories_Base_Controller.php]:
```suggestion
				'rest_cannot_create',
```

File: packages/dashboard/src/app/api/useStoryApi.js [hunk 4]
```diff
@@ -127,33 +121,17 @@ const useStoryApi = (dataAdapter, { storyApi, encodeMarkup }) => {
...
 
-        if (
-          cleanStories.length === 0 &&
-          cleanStories.length !== response.body.length
-        ) {
-          dispatch({
-            type: STORY_ACTION_TYPES.FETCH_STORIES_FAILURE,
-            payload: {
-              message: ERRORS.LOAD_STORIES.INCOMPLETE_DATA_MESSAGE,
```
Comment (swissspidy) [inline packages/dashboard/src/app/api/useStoryApi.js:148]: Let's also remove the actual `INCOMPLETE_DATA_MESSAGE` message as it's now unneeded.

File: packages/dashboard/src/app/api/useStoryApi.js [hunk 5]
```diff
@@ -20,9 +20,7 @@
 import { useCallback, useMemo, useReducer, useRef } from 'react';
 import { useFeatures } from 'flagged';
 import { addQueryArgs } from '@web-stories-wp/design-system';
-import { __, sprintf } from '@web-stories-wp/i18n';
 import { getTimeTracker } from '@web-stories-wp/tracking';
-import { base64Encode } from '@web-stories-wp/story-editor';
```
Comment (swissspidy) [inline packages/dashboard/src/app/api/useStoryApi.js:30]: Nice!

**Comment by swissspidy** [comment]
Some QA instructions and/or E2E tests (see #5043) would be great to prevent regressions.

File: includes/REST_API/Stories_Base_Controller.php (hunk 3 above)
Comment (spacedmonkey) [reply includes/REST_API/Stories_Base_Controller.php]: Fixed.

File: packages/dashboard/src/app/api/useStoryApi.js (hunk 4 above)
Comment (spacedmonkey) [reply packages/dashboard/src/app/api/useStoryApi.js:148]: Done.

File: packages/dashboard/src/app/api/test/apiProvider.js [hunk 6]
```diff
@@ -55,18 +53,20 @@ jest.mock('../wpAdapter', () => ({
       ],
     }),
   post: (path, { data }) => {
-    const title = typeof data.title === 'string' ? data.title : data.title.raw;
-    const content =
-      typeof data.content === 'string' ? data.content : data?.content?.raw;
-    const id = data.id || 456;
+    let title = '';
+    let id = data.id || 456;
+    if (data.original_id) {
```
Comment (spacedmonkey) [inline packages/dashboard/src/app/api/test/apiProvider.js]:
```suggestion
    if (data?.original_id) {
```

**Review by swissspidy — APPROVED** [review]
Some QA instructions and/or E2E tests (see #5043) would be great to prevent regressions. Otherwise LGTM

**Comment by swissspidy** [comment]
@googlebot I consent

**Pull request merged**